source/portfolio/portfolio_mad.cpp

```diff
@@ -0,0 +1,69 @@
+//
+// Created by eduar on 15/04/2021.
+//
+
+#include "portfolio_mad.h"
+#include <iostream>
+#include <numeric>
+namespace portfolio {
+    std::default_random_engine portfolio_mad::generator_ =
```

Comment:
Tirar do espaço global. Pode criar como static dentro da função que for usar. Se veio daquele exemplo do GA.h, realmente pode ter muita prática ruim lá porque o código já está bastante antigo.

Reply:
Removi do escopo global e declarei localmente nas funções.